src/test/mir-opt/dest-prop/cycle.main.DestinationPropagation.diff

```diff
@@ -56,7 +56,7 @@
           StorageLive(_6);                 // scope 3 at $DIR/cycle.rs:14:10: 14:11
 -         _6 = _1;                         // scope 3 at $DIR/cycle.rs:14:10: 14:11
 +         _6 = _4;                         // scope 3 at $DIR/cycle.rs:14:10: 14:11
-          _5 = const ();                   // scope 4 at $SRC_DIR/core/src/mem/mod.rs:LL:COL
+          _5 = const ();                   // scope 4 at $DIR/cycle.rs:14:5: 14:12
```

Comment:
Does this affect the generated debuginfo? LLVM inlining preserves the original source position instead of using the position of the inlined call. The MIR inliner should do the same.

Reply:
The debuginfo part was handled in https://github.com/rust-lang/rust/pull/68965

This here just addresses spans for diagnostics. These spans do preserve the original and all intermediate source positions across any level of inlining. I should probably adjust mir dumps to also print all spans instead of just the final one.